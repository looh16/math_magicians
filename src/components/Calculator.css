body {
  margin: 100px;
  display: flex;
  justify-content: center;
}

.wrapper {
  display: grid;
  grid-gap: 1px;
  grid-template-columns: repeat(4, [col] 100px);
  grid-template-rows: repeat(4, [row] auto);
  background-color: #fff;
  color: #444;
}

.box {
  background-color: #444;
  color: #fff;
  border-radius: 5px;
  padding: 20px;
  font-size: 150%;
}

.left {
  background-color: orange;
  text-align: center;
}

.right {
  background-color: lightgray;
  text-align: center;
}

.a {
  background-color: rgb(155, 147, 147);
  grid-column: col / span 4;
  grid-row: row;
  text-align: right;
}

.vd {
  grid-column: col / span 2;
  grid-row: row 5;
}
