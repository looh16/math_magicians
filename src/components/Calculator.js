import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [objCalc, setobjCalc] = useState({ total: 0, next: null, operation: null });

  const calculatorClickHandle = (e) => {
    setobjCalc((state) => calculate(state, e.target.innerText));
  };

  return (
    <section className="main-container">
      <div>
        <p>Let do some math</p>
      </div>
      <div className="container">
        <p className="box a">
          <span>{objCalc.total}</span>
          <span>{objCalc.operation}</span>
          <span>{objCalc.next}</span>
        </p>
        <button type="button" className="box right" onClick={calculatorClickHandle}>AC</button>
        <button type="button" className="box right" onClick={calculatorClickHandle}>+/-</button>
        <button type="button" className="box right" onClick={calculatorClickHandle}>%</button>
        <button type="button" className="box left" onClick={calculatorClickHandle}>รท</button>
        <button type="button" className="box right" onClick={calculatorClickHandle}>7</button>
        <button type="button" className="box right" onClick={calculatorClickHandle}>8</button>
        <button type="button" className="box right" onClick={calculatorClickHandle}>9</button>
        <button type="button" className="box left" onClick={calculatorClickHandle}>x</button>
        <button type="button" className="box right" onClick={calculatorClickHandle}>4</button>
        <button type="button" className="box right" onClick={calculatorClickHandle}>5</button>
        <button type="button" className="box right" onClick={calculatorClickHandle}>6</button>
        <button type="button" className="box left" onClick={calculatorClickHandle}>-</button>
        <button type="button" className="box right" onClick={calculatorClickHandle}>1</button>
        <button type="button" className="box right" onClick={calculatorClickHandle}>2</button>
        <button type="button" className="box right" onClick={calculatorClickHandle}>3</button>
        <button type="button" className="box left" onClick={calculatorClickHandle}>+</button>
        <button type="button" className="box zero right" onClick={calculatorClickHandle}>0</button>
        <button type="button" className="box right" onClick={calculatorClickHandle}>.</button>
        <button type="button" className="box left" onClick={calculatorClickHandle}>=</button>
      </div>
    </section>

  );
};

export default Calculator;
