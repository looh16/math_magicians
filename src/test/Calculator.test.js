/* eslint-disable no-undef */
import React from 'react';
import Renderer from 'react-test-renderer';
import Calculator from '../Components/Calculator';

it('renders correctly Calculator Component', () => {
  const tree = Renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
