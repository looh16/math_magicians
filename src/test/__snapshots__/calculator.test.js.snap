// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Calculator Component  1`] = `
<section
  className="main-container"
>
  <div>
    <p>
      Let do some math
    </p>
  </div>
  <div
    className="container"
  >
    <p
      className="box a"
    >
      <span>
        0
      </span>
      <span />
      <span />
    </p>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="box left"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="box left"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="box left"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="box left"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="box zero right"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="box right"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="box left"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</section>
`;
